packages/flutter_tools/lib/src/vmservice.dart

```diff
@@ -254,17 +262,23 @@ vm_service.VmService setUpVmService(
         }
       };
     });
-    vmService.registerService('flutterGetSkSL', 'Flutter Tools');
+    registrationRequests.add(vmService.registerService('flutterGetSkSL', 'Flutter Tools'));
   }
   if (printStructuredErrorLogMethod != null) {
     try {
-      vmService.streamListen(vm_service.EventStreams.kExtension);
+      await vmService.streamListen(vm_service.EventStreams.kExtension);
     } on vm_service.RPCError {
       // It is safe to ignore this error because we expect an error to be
       // thrown if we're already subscribed.
     }
     vmService.onExtensionEvent.listen(printStructuredErrorLogMethod);
   }
+
+  try {
+    await Future.wait(registrationRequests);
```

Comment:
Technically the return type of these methods is vm_service.Success, so either it works or we get an RPCError